src/core/app_messages.tf

```
@@ -0,0 +1,290 @@
+  count    = var.app_messages_enabled ? 1 : 0
+  name     = format("%s-io-p-app-messages-rg-01", local.project)
+  location = var.location
+
+  tags = var.tags
+}
+
+resource "azurerm_resource_group" "io-p-app-messages-rg-02" {
+  count    = var.app_messages_enabled ? 1 : 0
+  name     = format("%s-io-p-app-messages-rg-02", local.project)
+  location = var.location
+
+  tags = var.tags
+}
+
+locals {
+  app_messages = {
+    app_settings_common = {
+      FUNCTIONS_WORKER_RUNTIME       = "node"
+      WEBSITE_NODE_DEFAULT_VERSION   = "14.16.0"
+      WEBSITE_RUN_FROM_PACKAGE       = "1"
+      FUNCTIONS_WORKER_PROCESS_COUNT = 4
+      NODE_ENV                       = "production"
+
+      COSMOSDB_NAME                = "db"
+      COSMOSDB_URI                 = data.azurerm_cosmosdb_account.cosmos_api.endpoint
+      COSMOSDB_KEY                 = data.azurerm_cosmosdb_account.cosmos_api.primary_master_key
+      COSMOS_API_CONNECTION_STRING = format("AccountEndpoint=%s;AccountKey=%s;", data.azurerm_cosmosdb_account.cosmos_api.endpoint, data.azurerm_cosmosdb_account.cosmos_api.primary_master_key)
+
+      QueueStorageConnection = dependency.storage_account.outputs.primary_connection_string
+      MESSAGE_CONTAINER_NAME = dependency.storage_container_message-content.outputs.name
+
+      // Keepalive fields are all optionals
+      FETCH_KEEPALIVE_ENABLED             = "true"
+      FETCH_KEEPALIVE_SOCKET_ACTIVE_TTL   = "110000"
+      FETCH_KEEPALIVE_MAX_SOCKETS         = "40"
+      FETCH_KEEPALIVE_MAX_FREE_SOCKETS    = "10"
+      FETCH_KEEPALIVE_FREE_SOCKET_TIMEOUT = "30000"
+      FETCH_KEEPALIVE_TIMEOUT             = "60000"
+
+      FN_APP_STORAGE_CONNECTION_STRING = dependency.storage_account_app.outputs.primary_connection_string
+
+      // Events configs
+      EventsQueueStorageConnection = dependency.storage_account_apievents.outputs.primary_connection_string
+      EventsQueueName              = "events" # reference to https://github.com/pagopa/io-infra/blob/12a2f3bffa49dab481990fccc9f2a904004862ec/src/core/storage_apievents.tf#L7
+    }
+    app_settings_01 = {
+
+    }
+    app_settings_02 = {
+    }
+  }
+}
+
+# Subnet to host app messages function
+module "io-p-app-messages_01_snet" {
+  count                                          = var.app_messages_enabled && var.cidr_subnet_appmessages01 != null ? 1 : 0
+  source                                         = "git::https://github.com/pagopa/azurerm.git//subnet?ref=v1.0.51"
+  name                                           = format("%s-io-p-app-messages_01_snet", local.project)
+  address_prefixes                               = var.cidr_subnet_appmessages01
+  resource_group_name                            = azurerm_resource_group.rg_vnet.name
+  virtual_network_name                           = module.vnet.name
+  enforce_private_link_endpoint_network_policies = true
+
+  service_endpoints = [
+    "Microsoft.Web",
+  ]
```

```suggestion
  service_endpoints = [
    "Microsoft.Web",
    "Microsoft.AzureCosmosDB",
    "Microsoft.Storage",
  ]
```
add AzureCosmosDB and Storage